packages/apps/plugins/plugin-excalidraw/src/components/SketchComponent/SketchComponent.tsx

```diff
@@ -23,27 +23,65 @@ export type SketchComponentProps = {
   grid?: SketchGridType;
 };
 
+// TODO(burdon): Factor out.
+const findAncestorBySelector = (node: Element, selector: string): Element | null => {
+  while (node) {
+    if (node.matches(selector)) {
+      return node;
+    }
+    node = node.parentElement!;
+  }
+
+  return null;
+};
+
 /**
  * https://docs.excalidraw.com/docs/@excalidraw/excalidraw/api/props
  */
 export const SketchComponent = ({ sketch, className }: SketchComponentProps) => {
+  const containerRef = useRef<HTMLDivElement>(null);
   const { themeMode } = useThemeContext();
-  const model = useModel(sketch);
   const [down, setDown] = useState<boolean>(false);
-  // const [excalidrawAPI, setExcalidrawAPI] = useState<any>(null);
+  const excalidrawAPIRef = useRef<ExcalidrawImperativeAPI>();
+  const adapter = useStoreAdapter(sketch, {
+    onUpdate: ({ elements }) => {
+      excalidrawAPIRef.current?.updateScene({ elements });
+    },
+  });
 
-  const handleSave = () => {
-    const modified = model.getModified(true);
-    if (modified.length) {
-      log.info('handleSave', { modified: modified.map(({ id, version }) => `${id}:${version}`).join(',') });
+  // TODO(burdon): The mouse position gets offset within the deck, so we trigger a resize, which resets the component.
+  //  https://github.com/excalidraw/excalidraw/issues/7312
+  //  https://github.com/excalidraw/excalidraw/blob/master/packages/excalidraw/components/App.tsx
+  useEffect(() => {
+    const flash = () => {
+      setTimeout(() => {
+        window.dispatchEvent(new Event('resize'));
+      });
+    };
+
+    flash();
+    const deck = findAncestorBySelector(containerRef.current!, 'article')?.parentElement;
+    if (!deck) {
+      return;
     }
+
+    // Detect moved in deck.
+    const observer = new MutationObserver(() => flash());
+    observer.observe(deck, { attributes: false, childList: true, subtree: false });
+    return () => observer.disconnect();
+  }, []);
+
+  // Menu action.
+  const handleRefresh = () => {
+    // excalidrawAPIRef.current?.setToast({ message: 'Refresh' });
+    excalidrawAPIRef.current?.updateScene({ elements: adapter.getElements() });
```

Comment:
what does this do? I thought I saw in a video you shared that this was live, so why is a refresh button needed?

Reply:
there's no refresh button -- this is triggered on pointer up (i.e., after drag).